core/src/main/java/tc/oc/pgm/modules/ProjectileTrailMatchModule.java

```diff
@@ -0,0 +1,110 @@
+import tc.oc.pgm.events.ListenerScope;
+
+@ListenerScope(MatchScope.RUNNING)
+public class ProjectileTrailMatchModule implements MatchModule, Listener {
+
+  private static final String TRAIL_META = "projectile_trail_color";
+  private static final String CRITICAL_META = "arrow_is_critical";
+
+  private final Match match;
+
+  public ProjectileTrailMatchModule(Match match) {
+    this.match = match;
+    match
+        .getExecutor(MatchScope.RUNNING)
+        .scheduleAtFixedRate(this::checkMatchProjectiles, 0l, 50, TimeUnit.MILLISECONDS);
+  }
+
+  public void checkMatchProjectiles() {
+    match.getWorld().getEntitiesByClass(Projectile.class).stream()
+        .filter(projectile -> projectile.hasMetadata(TRAIL_META))
+        .forEach(
+            projectile -> {
+              if (projectile.isDead() || projectile.isOnGround()) {
+                projectile.removeMetadata(TRAIL_META, PGM.get());
+              } else {
+                final Color color = (Color) projectile.getMetadata(TRAIL_META, PGM.get()).value();
+
+                match.getPlayers().stream()
+                    .filter(
+                        pl ->
+                            pl.getSettings().getValue(SettingKey.EFFECTS)
+                                == SettingValue.EFFECTS_ON)
+                    .forEach(
+                        player -> {
+                          player
+                              .getBukkit()
+                              .spigot()
+                              .playEffect(
+                                  projectile.getLocation(),
+                                  Effect.COLOURED_DUST,
+                                  0,
+                                  0,
+                                  rgbToParticle(color.getRed()),
+                                  rgbToParticle(color.getGreen()),
+                                  rgbToParticle(color.getBlue()),
+                                  1,
+                                  0,
+                                  50);
+                        });
+              }
+            });
+  }
+
+  private float rgbToParticle(int rgb) {
+    return Math.max(0.001f, (rgb / 255.0f));
+  }
+
+  @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+  public void onProjectileLaunch(ProjectileLaunchEvent event) {
+    MatchPlayer player = match.getPlayer(event.getActor());
+    if (player != null) {
+      final Projectile projectile = event.getEntity();
+      projectile.setMetadata(
+          TRAIL_META, new FixedMetadataValue(PGM.get(), player.getParty().getFullColor()));
+      // Set critical metadata to false in order to remove default particle trail.
+      // The metadata will be restored just before the arrow hits something.
+      if (projectile instanceof Arrow) {
+        final Arrow arrow = (Arrow) projectile;
+        arrow.setMetadata(CRITICAL_META, new FixedMetadataValue(PGM.get(), arrow.isCritical()));
+        arrow.setCritical(false);
```

Comment:
Yeah I'm not sure, maybe instead of making it a toggle per player, perhaps it could be a server-wide config value. I'm not sure if a white trail would be the right call, since the point of the toggle is to disable the effects. Uh any suggestions you have would be welcome.

Reply:
I was trying to say you could "simulate" a critical effect? `Effect.CRIT`

Reply:
I don't think the crits can have colors, so you could just replace `Effect.COLORED_DUST` with the crit one for players with the setting toggled off?

Reply:
How does it look with both the crit and color? 
Maybe simulate the critical effect as king says but to a less extent than vanilla does.

Reply:
Oh that’s actually a really cool idea. I’ll experiment with that later and let you guys know the results.

Reply:
Thanks again for the suggestion @KingOfSquares & @Pugzy. 
I adjusted it and works flawlessly 👍 Much better alternative to having them not show up at all.